### PR TITLE
[com_fields] Allow to override the default value label and description

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #14701.

### Summary of Changes
Mostly the default value needs some type specific information. This pr adds support to override the default value label and description trough plugin language strings without overriding the whole default value form field.

It changes the user field override to that new way and adds some language specific strings for the user group field.

### Testing Instructions
- Create a usergroup custom field.
- Create a user field.

### Expected result
- The default value has some type specific label and tooltip.
- The default value has some type specific label and tooltip.

### Actual result
- The default tooltip is displayed.
- The default value has some type specific label and tooltip.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini [hunk 1]
```diff
@@ -4,6 +4,8 @@
 ; Note : All ini files need to be saved as UTF-8
 
 PLG_FIELDS_USERGROUPLIST="Fields - Usergrouplist"
+PLG_FIELDS_USERGROUPLIST_DEFAULT_VALUE_DESC="A comma separated list of user group ids."
```
Comment (infograf768) [inline administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini]: My tests show that this accept only one id. It does not work with multiple ids, whatever the setting for Multiple.

**Comment by laoneo** [comment]
Try it with my last commit, it should work now with multiple ids.

**Comment by infograf768** [comment]
@laoneo 
But what would happen if Multiple is set to No?

**Comment by laoneo** [comment]
Guess only the first item is selected.

**Comment by infograf768** [comment]
will test in afew

File: administrator/components/com_fields/models/field.php [hunk 2]
```diff
@@ -764,6 +764,19 @@ protected function preprocessForm(JForm $form, $data, $group = 'content')
 			{
 				$form->setFieldAttribute('type', 'readonly', 'true');
 			}
+
+			// Allow to override the default value label and description trough the plugin
```
Comment (brianteeman) [inline administrator/components/com_fields/models/field.php]: ocd kicking in

through not trough

File: libraries/cms/form/field/usergrouplist.php [hunk 3]
```diff
@@ -35,6 +35,29 @@ class JFormFieldUserGroupList extends JFormFieldList
 	protected static $options = array();
 
 	/**
+	 * Method to attach a JForm object to the field.
+	 *
+	 * @param   SimpleXMLElement  $element  The SimpleXMLElement object representing the `<field>` tag for the form field object.
+	 * @param   mixed             $value    The form field value to validate.
+	 * @param   string            $group    The field name group control value. This acts as as an array container for the field.
```
Comment (brianteeman) [inline libraries/cms/form/field/usergrouplist.php]: please remove the second as

Comment (brianteeman) [inline libraries/cms/form/field/usergrouplist.php]: please remove the second as

**Comment by laoneo** [comment]
@brianteeman done.

File: administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini [hunk 4]
```diff
@@ -4,6 +4,8 @@
...
+PLG_FIELDS_USERGROUPLIST_DEFAULT_VALUE_LABEL="Default Usergroups"
```
Comment (infograf768) [inline administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini]: I guess rather "Default User Groups"

**Comment by infograf768** [comment]
> Guess only the first item is selected.

Tested: In fact it is the last one selected

File: administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini (hunk 4 above)
Comment (laoneo) [inline administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini]: In `PLG_FIELDS_USERGROUPLIST` it is also written as one word and it passed the Brian filter :smile:.

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini]: My mistake missing it - @infograf768 is correct it is two words https://developer.joomla.org/en-gb-user-interface-text-guidelines/words-to-watch.html

Comment (laoneo) [inline administrator/language/en-GB/en-GB.plg_fields_usergrouplist.ini]: Right, below it is also written with two words. Changed it.

**Comment by laoneo** [comment]
@infograf768 can you please mark it as successful test in the tracker. Thanks.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on fbc29d8a39443da867befbd293aa69f50cb7e31c<br /><br />At Field 

- `User` Value and Tooltip stay with or -out PR same,
- `Usergroup` Value and Tooltip change specific for Type.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14793">issues.joomla.org/tracker/joomla-cms/14793</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on fbc29d8a39443da867befbd293aa69f50cb7e31c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14793">issues.joomla.org/tracker/joomla-cms/14793</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC as there are 2 successfully Tests?

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14793">issues.joomla.org/tracker/joomla-cms/14793</a>.</sub>

**Pull request merged**